### PR TITLE
Error messages for the contract validations show code/no validation message

### LINKED ISSUE #223: Add custom token: Error messages for the contract validations show code/no validation message about unsuccessful attempt of unlocking an account
UPD: 17/09/2020
Validation messages are not displayed at all not on the pop-uo, however, they are displayed in console
image

Related to #202

1. Error messages for the contract validations show code: see the video: 
https://drive.google.com/file/d/1UWUkgwc-cHCm_I1pQiU-4sijvPLwVBPK/view
**Expected result:** messages are written in understandable to an end user manner

2. No validation message about unsuccessful attempt of unlocking an account (see the video)
 https://drive.google.com/file/d/1kiGBF0UK4GaG6wDyXEEYXKZwF65DgOoT/view

**Expected result:** Error message should appear on the window

### PR DESCRIPTION
**Pull request by davidalbela**
Closes #223 

- Reduce message error from CustomCollateralModal for SplitPosition page.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr271--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr271--conditionaltokensexplorer.review.gnosisdev.com

**Comment by mariano-aguero** [comment]
Hi @davidalbela is this already done ? I see a task not checked, can you confirm ? thanks in advance

**Comment by davidalbela** [comment]
> Hi @davidalbela is this already done ? I see a task not checked, can you confirm ? thanks in advance

I would like to add the validation message before merge, thanks to you!

**Comment by mariano-aguero** [comment]
The second error 
```
No validation message about unsuccessful attempt of unlocking an account (see the video)
https://drive.google.com/file/d/1kiGBF0UK4GaG6wDyXEEYXKZwF65DgOoT/view
```
seems to be already fixed in the develop branch, just a clarification..

File: src/hooks/useCollateral.tsx [hunk 1]
```diff
@@ -47,6 +47,9 @@ export const useCollateral = (
         if (!cancelled) {
           setLoading(false)
           setCollateral(null)
+          if (err.message.indexOf('(') !== -1) {
+            err.message = err.message.split('(')[0]
```
Comment (mariano-aguero) [inline src/hooks/useCollateral.tsx:51]: Could you capitalize the first letter? Thank you

**Review by mariano-aguero — COMMENTED** [review]
See comments

**Review by hpmaxi — CHANGES_REQUESTED** [review]
Maybe some errors don't have a '(', or this changes in the future. Maybe you can add some function in `utils` that search for some patterns in the error.message or error.reason and return constant values likes the ones in `types.ts`, and default to something generic (and maybe logging that is and special case so we can handle it in future changes).

**Comment by elena-zh** [comment]
The 2nd issue seems to be unfixed now. 
I expected here to get a validation message something like 'MetaMask Tx Signature: User denied transaction signature.' or so. 
However, there is another task exists that is related to the Unlock contract section -  #280. 
I think that it will be possible to fix the 2nd case under the new task and show an error message when a user rejects transaction in Metamask or any other connected app.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr271--conditionaltokensexplorer.review.gnosisdev.com

**Comment by mariano-aguero** [comment]
@davidalbela any updates on the second error ?

**Review by hpmaxi — APPROVED** [review]

**Review by mariano-aguero — APPROVED** [review]

**Comment by davidalbela** [comment]
Can I refactor `getTokenSummary` method?

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr271--conditionaltokensexplorer.review.gnosisdev.com

**Comment by elena-zh** [comment]
Verified. 

The 1st case is fixed. 
As for the 2nd case: currently, there is no error message/unlock section does not change its state the error state https://zpl.io/aXeP0QE when a user rejects the unlock transaction in Metamask. The section returns to its 'Unlock' default view.
@mariano-aguero , could you please confirm that we should leave it as it is?

**Comment by davidalbela** [comment]
@elena-zh @mariano-aguero I will create other PR for 2nd case

**Pull request merged**